rq/batch.py

```diff
@@ -0,0 +1,85 @@
+import logging
+from typing import List, Union, Optional
+from uuid import uuid4
+
+from redis import Redis
+from redis.client import Pipeline
+
+from .job import Job
+from .utils import as_text
+
+logger = logging.getLogger("rq.job")
+
+
+class Batch:
+    REDIS_BATCH_NAME_PREFIX = 'rq:batch:'
+
+    def __init__(
+        self, id: str = str(uuid4())[-12:], jobs: List[Job] = None, connection: Optional['Redis'] = None, ttl=500
+    ):
+        self.id = id
+        self.connection = connection
+        self.key = '{0}{1}'.format(self.REDIS_BATCH_NAME_PREFIX, self.id)
+        self.jobs_key = self.key + ":jobs"
+        self.ttl = ttl
+
+        self.jobs = []
+        if jobs:
+            self.add_jobs(jobs)
+
+    def add_jobs(self, jobs: List[Union['Job', str]], pipeline=None):
+        pipe = pipeline if pipeline else self.connection.pipeline
+        self.job_ids = [job.id for job in jobs]
+        self.connection.sadd(self.jobs_key, *self.job_ids)
```

Comment:
Another consideration is what kind of data structure do we use to represent a batch in Redis? Options include:
* Stream - used to store job results
* SortedSet - used by job registries
* Set (current)

Reply:
I'm happy with the set. As you said, batch should just be a container for job IDs that gets cleaned out as those jobs expire.